main.py

```diff
@@ -38,52 +40,60 @@ def __init__(self):
-                self.database.log_error_causing_submission(None, submission, traceback.format_exc(), True)
+            if ExcludedArtists.is_excluded(parsed_submission):
                 continue
 
             for service in self.services:
                 try:
+                    # Some services can run without a successful parsed submission and just need the url
+                    # If this is not the case, it will throw an exception and continue
                     result = service.get_service_result(parsed_submission)
 
-                    if result.exceeds_threshold is True:
-                        self._perform_exceeds_threshold_mod_action(submission, result)
-                        exceeds_threshold = True
+                    if result.exceeds_threshold is False:
+                        continue
+
+                    self._perform_exceeds_threshold_mod_action(submission, result)
+                    exceeds_threshold = True
+
+                    break
 
-                        break
                 except Exception:
+                    # Go ahead and continue execution, don't want to fail completely just because one service failed.
                     traceback.print_exc(file=sys.stdout)
                     self.database.log_error_causing_submission(parsed_submission, submission, traceback.format_exc())
-                    # Go ahead and continue execution
-                    # Don't want to fail completely just because one service failed.
-                    pass
-                finally:
-                    self.database.save_submission(submission)
+
+            # submission is analyzed, saving
+            self.database.save_submission(submission)
+
+            if parsed_submission.success is False:
+                continue
 
             # Yeey this post probably isn't breaking the rules 🌈
             try:
                 if self.config.SEND_ARTIST_REPLY is True and exceeds_threshold is False:
                     self._reply_with_sticky_post(submission, self.last_fm.get_artist_reply(parsed_submission))
-            except (pylast.WSError, LookupError):
-                # Can't find artist, continue execution
+            except (pylast.WSError, EmptyDescription):
+                # Can't find artist or description is empty, logging
+                self.database.log_error_causing_submission(parsed_submission, submission, traceback.format_exc())
+
+    def _get_submissions(self):
+        for submission in self.reddit.subreddit(self.config.SUBREDDIT).stream.submissions():
+            print(submission.title + " - http://reddit.com" + submission.permalink)
+
+            if self.database.known_submission(submission) is True:
                 continue
 
+            if submission.is_self is True:
+                self.database.save_submission(submission)
+                # We currently don't do anything further with self posts.  Move to the next post.
+                continue
+
```

Comment:
Maybe we should just save every submission in this function before yielding it out and know we only need to log errors to the DB after this?

Reply:
yeah I like this